### PR TITLE
Implement consul and restructure peer

### PR DESCRIPTION
**Pull request by CeylonT**

### CODE REVIEW

**Comment by UtopiaBeam** [comment]
You use `gin` and not `echo`?

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  peer-service-0:
+    container_name: todo-peer
+    image: golang:1.15-alpine
+    restart: unless-stopped
+    volumes:
+      - ./peer:/app/peer
+    working_dir: /app/peer
+    command: sh -c 'go mod download && go run *.go'
+    environment: 
+      - USER_SERVICE_PORT=9000
```
Comment (UtopiaBeam) [inline docker-compose.yml]: This should be `PEER_SERVICE_PORT`

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  #   environment: 
+  #     - USER_SERVICE_PORT=9000
+  #     - CONSUL_HTTP_ADDR=consul:8500 
+  #     - REGISTRATION_ID=1
+  #   ports: 
+  #     - "9002:9000" 
+  #   depends_on:
+  #     - consul
+  #   networks:
+  #     - cdn-net
```
Comment (leo2tigers) [inline docker-compose.yml]: Maybe just delete this anyway instead of comment?

**Review by UtopiaBeam — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (CeylonT) [reply docker-compose.yml]: Leave it like that for testing purpose

File: docker-compose.yml (hunk 1 above)
Comment (CeylonT) [reply docker-compose.yml]: Already fixed

**Review by UtopiaBeam — APPROVED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (leo2tigers) [reply docker-compose.yml]: 👌

**Review by leo2tigers — APPROVED** [review]

**Pull request merged**